File: app/src/main/java/com/left/gank/ui/daily/DailyGirlContract.java
package com.left.gank.ui.daily;

import android.business.domain.Gift;
import android.business.domain.Girl;
import android.content.Context;

import com.left.gank.mvp.base.LoadMorePresenter;
import com.left.gank.mvp.base.SupportView;

import java.util.ArrayList;
import java.util.List;

import androidx.annotation.NonNull;


/**
 * Create by LingYan on 2016-10-26
 */

public interface DailyGirlContract {
    interface View extends SupportView {
        void refillData(List<Girl> list);

        void appendItem(List<Girl> list);

        void setMaxProgress(int value);

        void disProgressDialog();

        void openBrowseActivity(ArrayList<Gift> list);
    }

    abstract class Presenter extends LoadMorePresenter<View> {
        public Presenter(@NonNull Context context, View view) {
            super(context, view);
        }

        abstract void girlsImages(String url);
    }
}

File: app/src/main/java/com/left/gank/ui/daily/DailyGirlContract.kt
package com.left.gank.ui.daily

import android.business.domain.Gift
import android.business.domain.Girl
import android.content.Context

import com.left.gank.mvp.base.LoadMorePresenter
import com.left.gank.mvp.base.SupportView


/**
 * Create by LingYan on 2016-10-26
 */

interface DailyGirlContract {
    interface View : SupportView {
        fun loadDailyGirlSuccess(list: List<Girl>)

        fun loadDailyGirlFailure(msg: String)

        fun loadImagesSuccess(list: List<Gift>)

        fun loadImagesFailure(msg: String)

        fun showLoadingDialog()

        fun hideLoadingDialog()
    }

    abstract class Presenter(context: Context, view: View) : LoadMorePresenter<View>(context, view) {
        internal abstract fun loadGirls()

        internal abstract fun getImages(url: String)
    }
}
